app/HutchManager/Services/AgentService.cs

```diff
@@ -14,18 +15,21 @@ public AgentService(ApplicationDbContext db)
     _db = db;
   }
   
-  public async Task<Models.AgentModel> Create(Models.ManageAgent manageAgent)
+  public async Task<Models.AgentSummary> Create(Models.ManageAgent manageAgent)
   {
     Agent agent = new Agent()
     {
       Name = manageAgent.Name,
       ClientId = manageAgent.ClientId,
-      ClientSecretHash = manageAgent.ClientSecretHash.Sha256()
+      ClientSecretHash = manageAgent.ClientSecret.Sha256() // Has the secret
```

Comment:
I think this will change to do the id and secret generation for new agents, returning the unhashed values of each so the ui can display them one time straight after creation